protobuf/s12proto/validator_helpers.go

```diff
@@ -14,17 +14,24 @@ const (
 
 const (
 	uuid string = "^[0-9a-f]{8}-?[0-9a-f]{4}-?[0-9a-f]{4}-?[0-9a-f]{4}-?[0-9a-f]{12}$"
+	// legacyId contains the id without the document prefix; the prefix is accounted for in the service implementation code
+	legacyId string = "(?i)^[0-9a-f]{8}-?[0-9a-f]{4}-?[0-9a-f]{4}-?[0-9a-f]{4}-?[0-9a-f]{12}(-?[0-9a-f]{2,5}-?[0-9a-f]{16})?$"
 )
 
 var (
 	rxUUID = regexp.MustCompile(uuid)
+	rxLegacyId = regexp.MustCompile(legacyId)
 )
 
 // IsUUID checks if the string is a UUID (version 3, 4 or 5).
 func IsUUID(str string) bool {
 	return rxUUID.MatchString(str)
 }
 
+func IsLegacyID(str string) bool {
```

Comment:
The documentation should go on the exported function, rather than the private regex

Reply:
Moved the documentation to the exported function.